packages/agent-infra/mcp-client/src/StreamableHttpClient.ts

```diff
@@ -0,0 +1,402 @@
+/**
+ * The following code is modified based on
```

Comment:
What problems are solved by not using the official implementation [StreamableHTTPClientTransport](https://github.com/modelcontextprotocol/typescript-sdk/blob/main/src/client/streamableHttp.ts)?

Reply:
OAuth

Reply:
any related issue references here?

Reply:
OAuth authentication is not handled automatically by default https://blog.cloudflare.com/remote-model-context-protocol-servers-mcp/#making-authentication-and-authorization-just-work-with-mcp

Reply:
Oh, that's because version 1.9.0 of @modelcontextprotocol/sdk doesn't include [StreamableHTTPClientTransport](https://github.com/modelcontextprotocol/typescript-sdk/blob/main/src/client/streamableHttp.ts). Use a Fork before it's released.